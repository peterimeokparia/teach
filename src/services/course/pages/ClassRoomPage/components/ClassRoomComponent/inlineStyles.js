export const helpIconStyle = () => {
  return {
    fontSize: 70,
    position: "fixed",
    "z-index": 1, 
    left: "50%",
     top: "25%",
    // top: "-3%",
  };
};
