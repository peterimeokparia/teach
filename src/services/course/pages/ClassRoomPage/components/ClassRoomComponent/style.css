.ClassRoomDetail {
  height: 100vh;
  display: flex;
  flex-direction: column;
  font-family: 'Courier New', Courier, monospace;
}

.ClassRoomDetail form {
  display: flex;
  flex-direction: column;
  max-width: 300px;
  margin: 0;
}
.ClassRoomDetail form label {
  margin-bottom: 1em;
  font-weight: bold; 
}
.ClassRoomDetail form input {
  padding: 8px 6px 8px 8px 9px;
  border: 1px solid #aaa;
  border-radius: 2px;
}
.ClassRoomDetail form input:focus {
  outline: none;
  border: 1px solid #5d2faf;
}
.ClassRoomDetail form button {
  background: #5d2faf;
  font-size: 18px;
  color: #fff;
  padding: 10px;
  border: none;
  border-radius: 3px;
  box-shadow: 0 1px 2px #5d2faf;
  transition: box-shadow, background-color 0.12s;
  outline: none;
  margin-bottom: 10px;
}
.ClassRoomDetail form button + button {
  opacity: 0.8;
  font-size: 14px;
  padding: 7px;
  margin: initial 10px;
}
.ClassRoomDetail form button:hover {
  background-color: #753fd6;
  cursor: pointer;
}
.ClassRoomDetail form button[disabled] {
  opacity: 0.4;
}
.ClassRoomDetail form button:focus {
  box-shadow: 0 1px 8px #5d2faf;
}
.ClassRoomDetail form button:active {
  transform: scale(0.98);
}

.ClassRoomDetail header {
  background-color: #673ab7;
  color: #fff;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 10px 20px 20px;
  background-image: linear-gradient(to left, violet, red, white, green, yellow, orange, red);
}
.ClassRoomDetail header h1 {
  margin: 0;
  font-size: 24px;

   text-transform: uppercase;
    letter-spacing: 0.5rem;
    /* padding: 2rem 0; */
    font-weight: bold; 
    /* background-image: linear-gradient(to left, violet, red, white, green, yellow, orange, red); */
}
.ClassRoomDetail header .current-class {
  margin: 0;
  font-size: 16px;
  font-weight: 400;
  opacity: 0.8;
}
.header-left{
 /* margin-left: 10px; */
}

.header-right{
  margin-left: 40px;
}
.ClassRoomDetail header .current-class-button {
  margin: 0;
  font-size: 12px;
  font-weight: 400;
  opacity: 0.8;
}
.ClassRoomDetail .multiColor{
  /* background-color: #673ab7; */
  margin: 0;
  font-size: 24px;
  text-transform: uppercase;
  letter-spacing: 0.5rem;
  font-weight: bold;
  background-image: linear-gradient(to left, black, blue );
  -webkit-background-clip: text;
  -moz-background-clip: text;
  background-clip: text;
  box-shadow: 10px 10px black;
}
.ClassRoomDetail .content {
  display: flex;
  height: 100vh;
}
.ClassRoomDetail .sidebar {
  /* padding: 10px;  */
  padding-right: 0px;
  padding-left: 1px;
  display: flex;
  flex-direction: column;
}

.ClassRoomDetail .sidebar .lessons {
  list-style: none;
  margin: 0;
  padding: 0;
  top: 50%;
  left: 20%;
  margin-right: 1%;
}
.lessons .lessonMultiColor{
  background-color: #673ab7;
  margin: 0;
  /* font-size: 24px; */
  text-transform: uppercase;
  letter-spacing: 0.2rem;
  font-weight: bold;
  background-image: linear-gradient(to left, blue);
  -webkit-background-clip: text;
  -moz-background-clip: text;
  background-clip: text;
  color: transparent;
}
.class {
  margin-left: 280px;
  padding-top: 10px;
  height: 110vh;
}
.lesson2 {
  /* margin-left: 300px;
  padding-top: 10px;
  height: 110vh; */
  margin-left: 10px;
  width: 50vh; 
} 
.lessons .class-item {
  margin-bottom: 5px;
  border-style: solid;
  border-radius: 3px;
  top: 50%;
  left: 50%;
  margin-right: 0%;
  padding-bottom: 15px;
  padding-left: 0.5rem;

  text-transform: uppercase;
  letter-spacing: 0.2rem;
  /* padding: 0.5rem 0;  */
  font-weight: bold; 
}
.toggle-stage-btns-show,
.plan-class-btn,
.current-class-button,
.delete-class-btn,
.edit-class-btn {
  opacity: 0;
  border: none;
  font-size: 10px;
  cursor: pointer;
}
.toggle-stage-btns-show {
  visibility: visible;
  margin-left: 5px;
}
.plan-class-btn {
  margin-left: 5px;
}
.edit-class-btn {
  margin-right: 5px;
}
.delete-class-btn {
  margin-right: 5px;
}
.toggle-stage-btns-hide {
  visibility: hidden;
} 
.class-item:hover .toggle-stage-btns-show,
.toggle-stage-btns-show:focus,
.class-item:hover .plan-class-btn,
.plan-class-btn:focus,
.class-item:hover .current-class-button,
.current-class-button:focus,
.class-item:hover .delete-class-btn,
.delete-class-btn:focus,
.class-item:hover .edit-class-btn,
.edit-class-btn:focus {
  opacity: 1;
}
.class-item.selected {
  background: #e9e5ff;
  color: #222;
  font-weight: 600;
  box-shadow: 0 0 3px #6734ba, 0 0 3px inset #fff;
}
.class,
.class > div,
.class-content {
  display: flex;
  flex: 1;
  flex-direction: column;
  /* margin-left: 12%; */
  margin-left: 12%;
}
.class > div {
  padding: 10px;
  display: flex;
}
.class-editor {
  font-size: 18px;
  font-family: 'Fira Mono', monospace;
  flex: 1;
  padding: 10px;
}
.class-editor-help {
  display: flex;
  justify-content: space-between;
}
.preview-btn {
  margin: 10px;
  background-color: #fff;
  color: #555;
  font-size: 14px;
  border: none;
  border-radius: 2px;  
}
.preview-btn:active {
  background-color: #f7f2ff;
}
.span-btns {
  padding-left: 50px;
  padding-top: 0px;
}
.StudentDetailPageLink{
  padding-top: 20px;
  font-size: 18px;
  font-weight: 600;
}
.multiColor{
  /* background-color: #673ab7; */
  margin: 0;
  font-size: 24px;
  text-transform: uppercase;
  letter-spacing: 0.5rem;
  font-weight: bold;
  /* background-image: linear-gradient(to left, violet, red, white, green, yellow, orange, red); */
  -webkit-background-clip: text;
  -moz-background-clip: text;
  background-clip: text;
  color: transparent;
}
.viewGradesLink, .checkBoxViewGradesLink {
  color: blue;
  cursor: pointer;
}
.viewGradesLink, .checkBoxViewGradesLink:hover {
  color: #673ab7;
}
.checkBoxViewGradesLink {
  color: white;
  cursor: pointer;
}
.checkBoxViewGradesLink:hover {
  color: greenyellow;
}

/* unvisited link */
 a:link {
  color: blue;
}

/* visited link */
 /* a:visited {
  color: green;
}  */

/* mouse over link */
a:hover {
  color: #673ab7;
}

/* selected link */
/* a:active {
  color: blue;
} */ 

.invite-btn-animation {

  opacity: 1;
  border: 2px;
  font-size: 10px;
  cursor: pointer;
  padding-left: 30px;
  margin-top: 15px;
  padding-right: 30px;
  margin-left: 25px;
  box-shadow: 10px 10px;

  /* width: 100px;
  height: 100px; */
  background-color: #20c997; 
  position: relative;
  animation-name: example;
  animation-duration: 4s;
  animation-iteration-count: 2;
}

.AddStudentGradeComponent{
  margin-left: 12%;
}
.MarkAttendanceComponent{
  margin-left: 12%;
}

@media only screen and (min-width: 638px) {
  .ClassRoomDetail .content {
    flex-direction: row;
    height: 100%;
  }
  .ClassRoomDetail .sidebar {
    flex-basis: 300px;
    background: #fbf9ff;
    border-right: 1px solid #bba0ea;
  }
  .ClassRoomDetail .class {
    flex: 1;
  }
}
@media only screen and (max-width: 637px) {
  .ClassRoomDetail .content {
    flex-direction: column;
  }
  .ClassRoomDetail .sidebar {
    background: #fbf9ff;
    border-bottom: 1px solid #bba0ea;
    min-height: 50px;
  }
  .ClassRoomDetail .class {
    flex: 1;
  }
}
.ClassRoomDetail .gradesHeader{
  padding-left: 80px;
}
.ClassRoomDetail .grades{
  padding-left: 165px;
}

.flexible-modal {
  position: absolute;
  z-index: 1;
  border: 1px solid #ccc;
  background: white;
  /* padding-left: 1%; */
}
.flexible-modal-mask {
  position: fixed;
  height: 0%;
  background: rgba(55, 55, 55, 0.6);
  top:0;
  left:0;
  right:0;
  bottom:0;
}
.flexible-modal-resizer {
  position:absolute;
  right:0;
  bottom:0;
  cursor:se-resize;
  margin:5px;
  border-bottom: solid 2px #333;
  border-right: solid 2px #333;
}
.flexible-modal-drag-area{
  background: rgba(22, 22, 333, 0.2);
  height: 50px;
  position:absolute;
  right:0;
  top:0;
  cursor:move;
}
.tooltip {
  position: relative;
  display: inline-block;
  border-bottom: 1px dotted black;
}

.tooltip .tooltiptext {
  visibility: hidden;
  width: 120px;
  background-color: black;
  color: #fff;
  text-align: center;
  border-radius: 6px;
  padding: 5px 0;

  /* Position the tooltip */
  position: absolute;
  z-index: 1;
}

.tooltip:hover .tooltiptext {
  visibility: visible;
}


@keyframes example {
  0%   {background-color:red; left:0px; top:0px;}
  25%  {background-color:yellow; left:200px; top:0px;}
  50%  {background-color:blue; left:200px; top:200px;}
  75%  {background-color:green; left:0px; top:200px;}
  100% {background-color:red; left:0px; top:0px;}
}


@media (min-width: 1281px) {
  
  /* CSS */
  
}

/* 
  ##Device = Laptops, Desktops
  ##Screen = B/w 1025px to 1280px
*/

@media (min-width: 1025px) and (max-width: 1280px) {
  
  /* CSS */
  
}

/* 
  ##Device = Tablets, Ipads (portrait)
  ##Screen = B/w 768px to 1024px
*/

@media (min-width: 768px) and (max-width: 1024px) {
  
  /* CSS */
  
}

/* 
  ##Device = Tablets, Ipads (landscape)
  ##Screen = B/w 768px to 1024px
*/

@media (min-width: 768px) and (max-width: 1024px) and (orientation: landscape) {
  
  /* CSS */  
}

/* 
  ##Device = Low Resolution Tablets, Mobiles (Landscape)
  ##Screen = B/w 481px to 767px
*/

@media (min-width: 481px) and (max-width: 767px) {    
  /* CSS */
  .class,
  .class > div,
  .class-content {
    display: flex;
    flex: 1;
    flex-direction: column;
    /* margin-left: 12%; */
    margin-left: 30%;
  }
.AddStudentGradeComponent{
    margin-left: -30%;
  }
 .MarkAttendanceComponent{
    margin-left: -30%;
  }
}
/* 
  ##Device = Most of the Smartphones Mobiles (Portrait)
  ##Screen = B/w 320px to 479px
*/
@media (min-width: 320px) and (max-width: 480px) {
  /* CSS */
  .class,
  .class > div,
  .class-content {
    display: flex;
    flex: 1;
    flex-direction: column;
    /* margin-left: 12%; */
    margin-left: 39%;
  }
  .AddStudentGradeComponent{
    margin-left: -55%;
  }
  .MarkAttendanceComponent{
    margin-left: -55%;
  }
}