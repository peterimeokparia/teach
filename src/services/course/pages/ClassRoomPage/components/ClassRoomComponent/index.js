import { 
connect } from 'react-redux';

import { 
markAttendance } from 'services/course/actions/attendance';

import {
addNewGrade } from 'services/course/actions/grades';

import{
enableTeachPlatform } from 'services/course/actions/classrooms';

import { 
addNewGradesForSelectedStudents } from 'services/course/pages/GradesPage/components/AddStudentGrade/helpers';

import { 
markAttendanceForSelectedStudents } from 'services/course/pages/AttendancePage/components/MarkAttendanceComponent/helpers';

import { 
role } from 'services/course/helpers/PageHelpers';

import { 
navContent } from 'services/course/pages/components/NavigationHelper';

import {  
getUsersByOperatorId,    
getOperatorFromOperatorBusinessName,
getPushNotificationUsersByOperatorId,
getCoursesByOperatorId,
getSortedRecordsByDate } from 'services/course/selectors';

import { 
getLessonPlanUrls,
getselectedTutor, 
getStudentsSubscribedToCoursesByThisTutor, 
toggleBetweenAttendanceGradeDisplay,
emailInputOptions, 
emailMessageOptions } from  '../CourseLessonDropDownComponent/helpers';

import { 
openNewCourseModal, 
closeNewCourseModal } from 'services/course/actions/courses';

import { 
helpIconStyle } from './inlineStyles';
    
import NewCoursePage from 'services/course/pages/Courses/NewCoursePage';
import useClassRoomComponentHook from '../../hooks/useClassRoomComponentHook';
import MainMenu from 'services/course/pages/components/MainMenu';
import DropDown from 'services/course/pages/components/DropDown';
import CourseLessonDropDownComponent from 'services/course/pages/ClassRoomPage/components/CourseLessonDropDownComponent';
import LoginLogout from 'services/course/pages/LoginPage/components/LoginLogout';
import Roles from 'services/course/pages/components/Roles';
import CourseDetailCheckBoxComponent from 'services/course/pages/Courses/components/CourseDetailCheckBoxComponent';
import AddStudentGradeComponent from 'services/course/pages/GradesPage/components/AddStudentGrade';
import MarkAttendanceComponent from 'services/course/pages/AttendancePage/tests/MarkAttendanceComponent';
import MultiInputEmailComponent from 'services/course/pages/Email/MultiInputEmailComponent';
import Modal from 'react-modal';
import PostAddIcon from '@material-ui/icons/PostAdd';
import Swal from 'sweetalert2';
import './style.css';

const ClassRoomComponent = ({
    enableTeachPlatform,
    selectedUserId,
    operator,
    operatorBusinessName,
    currentUser,
    allSessions,
    users,
    grades,
    courses,
    markAttendance,
    addNewGrade,
    toggleSideBarDisplay,
    pushNotificationSubscribers,
    selectedCourseFromLessonPlanCourseDropDown, 
    isModalOpen,
    openNewCourseModal,
    closeNewCourseModal }) => {
    const selectedUser = getselectedTutor( users, selectedUserId );
    const currentGrades = grades?.filter(grd => grd?.courseId === selectedCourseFromLessonPlanCourseDropDown?._id);
    const url = getLessonPlanUrls( operatorBusinessName, selectedUserId );
    const sessions = allSessions?.filter( usersession => usersession?.courseId === selectedCourseFromLessonPlanCourseDropDown?._id);

    let {
        setListOfStudents,
        setDropDownDisplayOption,
        dropDownDisplayOption, 
        listOfStudents,
    } = useClassRoomComponentHook(operatorBusinessName, selectedUser, operator );

    const pushNotificationUsers = pushNotificationSubscribers?.filter(pushuser => listOfStudents?.find(student => student?._id === pushuser?.userId));
    
    function enableTeachCoursePlatform( ) {
        if ( ( !listOfStudents ) || ( listOfStudents?.length === 0 ) ) {
            Swal.fire({
                title: `Please invite students before starting a new teaching session.`,
                icon: 'warning',
                showCancelButton: true,
                confirmButtonText: 'Invite.',
                confirmButtonColor: '#673ab7',
                cancelButtonText: 'Not now.'
            }).then( (response) => {
              if ( response?.value ) {
                  return;
              } else { 
                  enableTeachPlatform({ listOfStudents, selectedTutorId: selectedUserId, operatorBusinessName, sessions, operator } );
              }
            }); 
    } else {
        enableTeachPlatform({ listOfStudents, selectedTutorId: selectedUserId, operatorBusinessName, sessions, operator });
    }
};

return (
    <div className="ClassRoomDetail"> 
            <header>
                <div>
                    <h1>  <span className="multiColor"> {selectedUser?.firstname} </span></h1> 
                    <span className="header-left">
                    <MainMenu 
                        navContent={navContent( currentUser, operatorBusinessName, currentUser?.role,  "Student" ).users}
                    />
                    </span>
                </div>
                <span>   
                    <button
                        className="plan-lesson-btn"
                        onClick={() => enableTeachCoursePlatform()}
                    >
                        { currentUser?.role === role.Tutor ? "Teach"  : "Join" }
                    </button>
                </span>  
                  <LoginLogout
                    operatorBusinessName={operatorBusinessName}
                    user={currentUser} 
                  />
            </header>
             <div className="content"> 
                <div className="sidebar">
                {( ( toggleSideBarDisplay ) || dropDownDisplayOption ) && (listOfStudents?.length > 0) && 
                    <Roles
                        role={currentUser?.role === role.Tutor }
                    >
                        <span className="left">   
                        <DropDown
                            label={""}
                            key={"_id"}
                            value={"name"}
                            optionCollection={toggleBetweenAttendanceGradeDisplay()}
                            setOptionSelectedValue={selectedOption => setDropDownDisplayOption(selectedOption)} 
                        />  
                        </span>     
                    </Roles> 
                }
                </div> 
                 <div className="class"> 
                {( courses?.filter( course => course?.createdBy === selectedUserId)?.length === 0 && ( currentUser?.role === role.Tutor ) ) 
                    ? <div> 
                        { ( currentUser?.role === role.Tutor ) && ( !isModalOpen ) &&
                            <PostAddIcon 
                                style={helpIconStyle()}
                                className="comment-round-button-5"
                                onClick={openNewCourseModal}
                            />
                        }
                        <Modal isOpen={isModalOpen} onRequestClose={closeNewCourseModal}> <NewCoursePage user={currentUser} operator={operator}/> </Modal>
                     </div> 
                    : <div>
                        {( dropDownDisplayOption === "Courses" || 
                            dropDownDisplayOption === "" ) &&  
                            <CourseLessonDropDownComponent
                                className="add-class-button"
                                operatorBusinessName={operatorBusinessName}
                                selectedUserId={selectedUserId}     
                            />
                        }
                        {( dropDownDisplayOption === "Grade" ) &&  
                            <div className="AddStudentGradeComponent"> 
                            <AddStudentGradeComponent
                                className="add-class-button"
                                selectedStudents={listOfStudents}
                                onSubmit={newGrade => addNewGradesForSelectedStudents( pushNotificationUsers, selectedCourseFromLessonPlanCourseDropDown, newGrade, currentGrades, addNewGrade  ) }
                            />
                            </div>    
                        }
                        {( dropDownDisplayOption === "Attendance" ) &&    
                            <div className="MarkAttendanceComponent">   
                            <MarkAttendanceComponent 
                                className="add-class-button"
                                selectedStudents={listOfStudents}
                                onSubmit={attendance => markAttendanceForSelectedStudents( pushNotificationUsers, selectedCourseFromLessonPlanCourseDropDown, attendance, markAttendance  ) }
                            />
                            </div>    
                        }
                    </div>
                }
                </div>
                <Roles
                    role={ currentUser?.role === role.Tutor }
                >
                    <div className="sidebar"> 
                        <CourseDetailCheckBoxComponent 
                            collection={(selectedCourseFromLessonPlanCourseDropDown) ? getStudentsSubscribedToCoursesByThisTutor( users, courses, selectedUserId )?.filter(student => student.courses.includes(selectedCourseFromLessonPlanCourseDropDown?._id)) : selectedCourseFromLessonPlanCourseDropDown}
                            setCollection={meetingInvitees => setListOfStudents(meetingInvitees)}
                            description={"firstname"}
                            operatorBusinessName={operatorBusinessName}
                        />     
                    </div>
                </Roles>
                <Roles
                    role={ currentUser?.role === role.Student }
                >
                <div className="sidebar"> 
                    <MultiInputEmailComponent
                        setLesson={selectedUser}
                        inputFieldOptions={emailInputOptions}
                        messageOptions={emailMessageOptions(currentUser, url?.lessonPageUrl)}
                    />
                </div>
                      </Roles>
                </div>
        </div>
    );
};

const mapDispatch = {
    enableTeachPlatform,
    openNewCourseModal,
    closeNewCourseModal,
    markAttendance, 
    addNewGrade,
};

const mapState = (state, ownProps) => {
    return {
        operator: getOperatorFromOperatorBusinessName(state, ownProps),
        pushNotificationSubscribers: getPushNotificationUsersByOperatorId(state, ownProps),
        currentUser: state.users.user,
        users: getUsersByOperatorId(state, ownProps),
        courses: getCoursesByOperatorId(state, ownProps),
        grades:  getSortedRecordsByDate(Object.values(state?.grades?.grades), 'testDate'),
        toggleSideBarDisplay: state.classrooms.displaySideBarDropDown,
        allSessions: Object.values(state?.sessions?.sessions),
        selectedCourseFromLessonPlanCourseDropDown: state.courses.selectedCourseFromLessonPlanCourseDropDown,
        isModalOpen: state?.courses?.isModalOpen
    };
};

export default connect( mapState, mapDispatch )(ClassRoomComponent);