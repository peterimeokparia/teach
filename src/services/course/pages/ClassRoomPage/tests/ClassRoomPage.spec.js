import React from 'react';
import ClassRoomComponent  from 'Services/course/Pages/ClassRoomPage/Components/ClassRoomComponent';
import renderer from 'react-test-renderer';
import thunk from "redux-thunk"; 
import * as reactRedux from 'react-redux';
import { Provider, useSelector } from "react-redux";
import { applyMiddleware } from 'redux'; 
import configureStore from 'redux-mock-store';

jest.mock('../../../Api');

const mockStore = configureStore([]);
  
let operatorBusinessName= "boomingllc";
let selectedUserId= "PERSON010010100000047";

describe('ClassRoomComponent', () =>  {  

 let  store = mockStore({
      operators: { operators: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} },
      notifications: { pushNotificationSubscribers : {_id:"5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" } },
      users: { users: {_id: "PERSON5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" }},
      courses: { courses: {_id: "COURSE5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" }},
      grades: { grades: {_id: "COURSE5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" }},
      classrooms: { classrooms: {displaySideBarDropDown: false }},
      sessions: { sessions: {_id: "SESSIONS5fcb0e19fd5e0117dc09dcfa", numberOfSessions: 9, totalNumberOfSessions: 1, status: true } },
      lessons: { lessons: { selectedLessonFromLessonPlanDropDown: "Test Lesson" } },
      courses: { courses: { selectedCourseFromLessonPlanCourseDropDown: "Test Course" }  },
      meetings: { meetings: {_id: "MEETING5fcb0e19fd5e0117dc09dcfa", numberOfSessions: 9, totalNumberOfSessions: 1, status: true }  },
  }, applyMiddleware(thunk));

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
 
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  it('renders as expected', async () => {

    const dispatch = jest.fn();
    useDispatchMock.mockReturnValue(dispatch)

    const tree = renderer.create(
      <Provider store={store}>
        <ClassRoomComponent 
          selectedUserId={selectedUserId}
          operatorBusinessName={operatorBusinessName } 
      /> 
    </Provider>
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});

