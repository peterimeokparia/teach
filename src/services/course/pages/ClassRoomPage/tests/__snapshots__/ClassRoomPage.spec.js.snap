// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassRoomComponent renders as expected 1`] = `
<div
  className="ClassRoomDetail"
>
  <header>
    <div>
      <h1>
          
        <span
          className="multiColor"
        >
           
           
        </span>
      </h1>
      <span
        className="header-left"
      >
        <div>
          <div
            className="burgerMenuDiv"
            onMouseDown={[Function]}
          >
            <div
              className="divOne"
            />
            <div
              className="divTwo"
            />
            <div
              className="divThree"
            >
               
            </div>
          </div>
          <div
            className="hide"
            id="flyoutMenu"
            onMouseUp={[Function]}
          >
            <li
              className="navlinkItem"
            >
              <a
                href="/boomingllc/mycourses"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                 
                My Courses
                 
              </a>
            </li>
            <li
              className="navlinkItem"
            >
              <a
                href="/boomingllc/courses"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                 
                All Courses
                 
              </a>
            </li>
            <li
              className="navlinkItem"
            >
              <a
                aria-current="page"
                href="/"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                 
                Tutors
                 
              </a>
            </li>
            <li
              className="navlinkItem"
            >
              <a
                href="/boomingllc/students"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                 
                Students
                 
              </a>
            </li>
          </div>
        </div>
      </span>
    </div>
    <span>
      <button
        className="plan-lesson-btn"
        onClick={[Function]}
      >
        Join
      </button>
    </span>
    <span
      className="lock-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
        />
      </svg>
      <div>
        <div>
          <div
            className="Toastify"
          />
        </div>
      </div>
    </span>
  </header>
  <div
    className="content"
  >
    <div
      className="sidebar"
    >
      
    </div>
    <div
      className="class"
    >
      <div>
        <div>
          <div
            className="dropdownComponents"
          >
            <span
              className="left"
            >
              <span
                className="dropDownSelector"
              >
                <span>
                  <form
                    onSubmit={[Function]}
                  >
                    <label>
                      <span
                        className="main-title"
                      >
                        Courses
                      </span>
                    </label>
                    <label>
                      <div
                        className="MuiInputBase-root MuiInput-root MuiInput-underline space-between-select"
                        onClick={[Function]}
                      >
                        <div
                          aria-haspopup="listbox"
                          aria-label="Without label"
                          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            dangerouslySetInnerHTML={
                              Object {
                                "__html": "&#8203;",
                              }
                            }
                          />
                        </div>
                        <input
                          aria-hidden={true}
                          className="MuiSelect-nativeInput"
                          onAnimationStart={[Function]}
                          onChange={[Function]}
                          tabIndex={-1}
                        />
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSelect-icon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                      </div>
                    </label>
                  </form>
                </span>
              </span>
            </span>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
